src/tracks/playabletrack/wavetrack/ui/WaveTrackAffordanceControls.cpp

```diff
@@ -22,6 +22,40 @@
 #include "../../../ui/AffordanceHandle.h"
 #include "WaveTrackView.h"//need only ClipParameters
 
+#include "../../../../ProjectHistory.h"
+#include "../../../../SelectionState.h"
+#include "../../../../RefreshCode.h"
+
+class WaveTrackAffordanceHandle final : public AffordanceHandle
+{
+public:
+    WaveTrackAffordanceHandle(const std::shared_ptr<Track>& track) : AffordanceHandle(track) { }
+
+    static UIHandlePtr HitAnywhere(std::weak_ptr<AffordanceHandle>& holder, const std::shared_ptr<Track>& pTrack)
+    {
+        auto result = std::static_pointer_cast<AffordanceHandle>(std::make_shared<WaveTrackAffordanceHandle>(pTrack));
+        result = AssignUIHandlePtr(holder, result);
+        return result;
+    }
+
+    UIHandle::Result SelectAt(const TrackPanelMouseEvent& event, AudacityProject* pProject) override
+    {
+        const auto track = std::dynamic_pointer_cast<WaveTrack>(TrackList::Get(*pProject).Lock<Track>(GetTrack()));
```

Comment:
This dynamic cast could safely be static instead.  WaveTrackAffordanceHandle can assume that it is only created by hit-tests on affordance areas of wave tracks.

Reply:
I don't like dymamic_cast, but for me it's a general rule to use it when downcasting is required, it spots places where refactoring may be needed.

Reply:
I disagree with that general rule.  static_cast is appropriate instead of dynamic_cast, when you have good reasons to tell the compiler, "trust me, I can prove that the check is not necessary."

dynamic_cast is also justifiable sometimes.